src/core/namespace.c

```diff
@@ -599,11 +599,26 @@ static int clone_device_node(const char *d, const char *temporary_mount) {
         dn = strjoina(temporary_mount, d);
 
         mac_selinux_create_file_prepare(d, st.st_mode);
-        r = mknod(dn, st.st_mode, st.st_rdev);
+        if (do_mknod)
+                r = mknod(dn, st.st_mode, st.st_rdev);
+        else
+                r = mknod(dn, S_IFREG, st.st_rdev);
```

Comment:
hmm, so the second argument controls both the node type and the access mode of it. You specify only the node type here, which means the access mode will be `0000`. Which is fine I guess, given that we overmount things right-away, and it probably is a good idea, if nobody can access the node below it. However, I think it would be better to make this explicit, i.e.

```c
r = mknod(dn, S_IFREG | 0000 /* no access! */, 0);
```

(Also, the third argument has no effect on S_IFREG, hence better pas it as 0...)

Reply:
adressed